### PR TITLE
Add Spark web application server

### PR DESCRIPTION
**Pull request by okucheruk**
This PR adds Spark web application server sample.

**Highlights:**
- application listening on `localhost:8080`;
- Gradle configuration refactored with using Groovy DSL;
- added Spark logger `slf4j` dependency;
- Gradle configuration extended with Google Guava and Truth libraries.
- variables of library versions are distributed with `gradle.properties` at root and module levels.

### CODE REVIEW

File: server/gradle.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+sparkCoreVersion = 2.8.0
```
Comment (Oleh-Stefanovych) [inline server/gradle.properties]: Let's define a version of libraries in the same place (`build.gradle` or `gradle.properties`).

Comment (okucheruk) [reply server/gradle.properties]: Refactored

**Comment by Oleh-Stefanovych** [comment]
@kuzmavictor, @denysLavrienov, PTAL.

File: server/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+plugins {
+    java
+}
+
+val sparkCoreVersion: String by project
+val slf4jApiVersion: String by extra
+
+dependencies {
+    compile("com.sparkjava:spark-core:$sparkCoreVersion")
+    compile("org.slf4j:slf4j-simple:$slf4jApiVersion")
```
Comment (kuzmavictor) [inline server/build.gradle.kts]: Please, delete redundant dependency.

**Comment by Oleh-Stefanovych** [comment]
@armiol, PTAL.

File: server/build.gradle.kts (hunk 2 above)
Comment (Oleh-Stefanovych) [reply server/build.gradle.kts]: Spark requires a SLF4J library:

See at the end of the page:
[http://sparkjava.com/documentation.html#examples-and-faq](http://sparkjava.com/documentation.html#examples-and-faq)

Comment (kuzmavictor) [reply server/build.gradle.kts]: You are not using a logger. The application works without it.

**Comment by armiol** [comment]
@okucheruk please add the Gradle-related changes to the PR description as well.

**Comment by okucheruk** [comment]
@armiol, fixed, PTAL.

File: server/build.gradle [hunk 3]
```diff
@@ -1 +1,14 @@
+/**
+ * Gradle build automation script written using a Groovy DSL
+ *
+ * Written by: Oleksii Kucheruk on Tue Oct 8 2019
```
Comment (armiol) [inline server/build.gradle]: As long as we don't use `@author` tags anywhere, please remove this line as well.

File: server/build.gradle [hunk 4]
```diff
@@ -1 +1,14 @@
+/**
+ * Gradle build automation script written using a Groovy DSL
```
Comment (armiol) [inline server/build.gradle]: This is obvious. Either say something meaningful or remove.

File: server/src/main/java/io/javaclasses/filehub/SparkJavaSample.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import static spark.Spark.get;
+
+/**
+ * Spark web application server sample.
+ */
+public class SparkJavaSample {
+
+    /**
+     * Demonstrates handling of GET HTTP request.
```
Comment (armiol) [inline server/src/main/java/io/javaclasses/filehub/SparkJavaSample.java]: Let's use `{@code GET}` as we discussed.

File: server/src/main/java/io/javaclasses/filehub/SparkJavaSample.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+     *
+     * <p>Handler responds with welcome message for HTTP GET requests to '/' URI.
+     * The server is listening on localhost:8080.
```
Comment (armiol) [inline server/src/main/java/io/javaclasses/filehub/SparkJavaSample.java]: Let's wrap the "localhost:8080" into `{@code ...}` as well.

File: server/build.gradle (hunk 3 above)
Comment (okucheruk) [reply server/build.gradle]: Removed

File: server/src/main/java/io/javaclasses/filehub/SparkJavaSample.java (hunk 5 above)
Comment (okucheruk) [reply server/src/main/java/io/javaclasses/filehub/SparkJavaSample.java]: Fixed

File: server/src/main/java/io/javaclasses/filehub/SparkJavaSample.java (hunk 6 above)
Comment (okucheruk) [reply server/src/main/java/io/javaclasses/filehub/SparkJavaSample.java]: Wrapped

**Comment by okucheruk** [comment]
@armiol, fixed, PTAL.

File: gradle.properties [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# Definitions of libraies versions for all sub-projects/modules
+slf4jVersion=1.7.28
+guavaVersion=28.1-jre
+truthCoreVersion=1.0
+
+# Change verbosity of warnings for tracking deprecated Gradle features
+org.gradle.warning.mode=all
```
Comment (armiol) [inline gradle.properties]: Please fix EOF.

File: server/gradle.properties [hunk 8]
```diff
@@ -0,0 +1,2 @@
+# Definitions of libraies versions for this module
+sparkCoreVersion = 2.8.0
```
Comment (armiol) [inline server/gradle.properties]: Please fix EOF here as well.

File: server/gradle.properties [hunk 9]
```diff
@@ -0,0 +1,2 @@
+# Definitions of libraies versions for this module
```
Comment (armiol) [inline server/gradle.properties]: `libraies` is not a word.

**Pull request merged**